internal/translator/translate.go

```diff
@@ -48,23 +49,28 @@ func IngressRouteToHTTPProxy(ir *irv1beta1.IngressRoute) (*hpv1.HTTPProxy, []str
 
 	// TODO(youngnick): Investigate if we should skip logically empty IngressRoutes
 
+	if ir.Spec.VirtualHost == nil {
```

Comment:
Suggestion: do this first, then you know from the point after this check, there can be no further paths that return an error.